spec.md

```diff
@@ -179,17 +179,163 @@ physically present in the final circuit. Refer to the description of the
 instance statement for details on how to instantiate a module
 ([@sec:instances]).
 
+## Optional Groups
+
+Optional groups are named collections of statements inside a module.  Optional
+groups contain optional functionality for a module that can be included or
+excluded without affecting the functionality of a circuit.  Optional groups are
+used to keep verification, debugging, or other collateral, not relevant to the
+operation of the circuit, in a separate area.  Each group can then be optionally
+included in the resulting design.
+
+The `declgroup`{.firrtl} keyword declares an optional group with a specific
+identifier.  An optional group may be declared in either a circuit, and external
+module, or in another optional group declaration.  An optional group's
```

Comment:
how do names work with groups declared in an external module?  do they alias and must have the same implementation as local?  Are they usable from the local circuit?  

I'm wondering if instead of adding bodies to external modules, we declare them using a declgroup root:
`extmodule Foo using declgroup A` with declgroup A being just a normal declaration.  Then there wouldn't be confusion about multiple conflicting As  (linking is then not just textural substitution, but with named types it already might not be).